File: internal/discovery/testdata/fixtures/typescript/src/agent.ts
import Anthropic from "@anthropic-ai/sdk";
import OpenAI from "openai";
import { generateText } from "ai";

const anthropic = new Anthropic();
const openai = new OpenAI();

async function classify(text: string) {
  return anthropic.messages.create({ model: "claude-sonnet-4-5", messages: [{ role: "user", content: text }] });
}
async function loopAgent(items: string[]) {
  for (const it of items) {
    await openai.chat.completions.create({ model: "gpt-4o", messages: [] });
  }
}
async function vercel() {
  return generateText({ model: openai("gpt-4o"), prompt: "summarize" });
}

File: internal/discovery/testdata/fixtures/kotlin/app/Agent.kt
package com.example.app

import dev.langchain4j.model.chat.ChatLanguageModel

class Agent(private val model: ChatLanguageModel) {
    fun classify(text: String): String {
        return model.generate(text)
    }

    fun batch(items: List<String>) {
        for (item in items) {
            model.generate(item)
        }
    }
}
